packages/davinci-client/src/lib/collector.types.ts

```diff
@@ -76,18 +76,25 @@ export interface ActionCollectorNoUrl<T extends ActionCollectorTypes> {
   };
 }
 
+type ActionCollectorOutputTypes =
+  | {
+      key: string;
+      label: string;
+      type: string;
+      url: string;
+    }
+  | {
+      key: string;
+      label: string;
+      type: string;
+    };
 export interface ActionCollectorWithUrl<T extends ActionCollectorTypes> {
   category: 'ActionCollector';
   error: string | null;
   type: T;
   id: string;
   name: string;
-  output: {
-    key: string;
-    label: string;
-    type: string;
-    url: string | null;
-  };
+  output: ActionCollectorOutputTypes;
```

Comment:
I'm not entirely sure what we are doing here. This `interface` is named `ActionCollectorWithUrl`, but the output may or may not have a `url`. That seems a bit wonky.

May I ask why we are going to such lengths to avoid just having a single type that has `url` as an optional property? I've noticed this complexity, and I'm just not sure what value we get from it. But, I'm probably missing something.